packages/cosmo-dycore/package.py

```diff
@@ -97,31 +90,18 @@ class CosmoDycore(CMakePackage):
             default='gpu',
             description='Slurm constraints for nodes requested')
 
-    depends_on('gridtools@1.1.3 ~cuda', when='~cuda+gt1')
-    depends_on('gridtools@1.1.3 +cuda', when='+cuda+gt1')
+    depends_on('gridtools@1.1.3 ~cuda', when='~cuda +gt1')
+    depends_on('gridtools@1.1.3 +cuda', when='+cuda +gt1')
     depends_on('boost@1.65.1: +program_options +system')
-    depends_on('serialbox@2.6.0', when='+build_tests', type=('run'))
-    depends_on('mpi', type=('build', 'link', 'run'), when='~cuda')
-    depends_on('mpi +cuda', type=('build', 'link', 'run'), when='+cuda')
+    depends_on('serialbox@2.6.0', when='+build_tests', type='run')
```

Comment:
I changed serialbox to a run dep. But I think serialbox should be a build,link and run dep, since we link against the c-libs.

Reply:
This PR intentionally doesn't change serialbox's dependency type. But I think we should tackle this problem in an other PR 👍